packages/eui/src/components/modal/modal.tsx

```diff
@@ -62,9 +63,15 @@ export const EuiModal: FunctionComponent<EuiModalProps> = ({
 }) => {
   const onKeyDown = (event: React.KeyboardEvent<HTMLDivElement>) => {
     if (event.key === keys.ESCAPE) {
-      event.preventDefault();
-      event.stopPropagation();
-      onClose(event);
+      if (
+        isDOMNode(event.target) &&
+        (event.target === event.currentTarget ||
+          event.currentTarget.contains(event.target))
+      ) {
```

Comment:
Amazingly elegant approach - huge kudos! 👏 

FYI that [`.contains()` is inclusive](https://developer.mozilla.org/en-US/docs/Web/API/Node/contains) so you can drop the extra `target === currentTarget` check.

```suggestion
      if (
        isDOMNode(event.target) &&
        event.currentTarget.contains(event.target)
      ) {
```

Reply:
Yes, of course you're right, I always forget about it because the naming is not suggesting it 😅 
Removed in this [commit](https://github.com/elastic/eui/pull/7939/commits/bb505879ebc9a30dfcb8e841f5f2e7ac53231bd5).